advanced_source/visualizing_gradients_tutorial.py

```diff
@@ -0,0 +1,593 @@
+"""
+Visualizing Gradients
+=====================
+
+**Author:** `Justin Silver <https://github.com/j-silv>`__
+
+When training neural networks with PyTorch, it’s possible to ignore some
+of the library’s internal mechanisms. For example, running
+backpropagation requires a simple call to ``backward()``. This tutorial
+dives into how those gradients are calculated and stored in two
```

Comment:
Maybe something more like "This tutorial explains the subtleties of requires_grad, retains_grad, leaf and non-leaf using a simple example..."

since I feel like that is the highlight of the tutorial rather than "dives into how those gradients are calculated"